Source/CsDebugScript.Engine/Engine/Debuggers/DbgEngDll.cs

```diff
@@ -75,6 +75,23 @@ internal class DbgEngDll : IDebuggerEngine
         private static IDebugSystemObjects4 systemObjects;
 
         /// <summary>
+        /// Dictionary of all debugee flow controlers. Key is process id that is being debugged.
+        /// </summary>
+        private static DictionaryCache<uint, DebuggeeFlowController> debugeeFlowControlers;
+
+        /// <summary>
+        /// Static constructor.
+        /// </summary>
+        static DbgEngDll()
+        {
+            // Populate flow controlers lazely.
+            //
+            debugeeFlowControlers =
+                new DictionaryCache<uint, DebuggeeFlowController>(
+                    (processId) => new DebuggeeFlowController(originalClient));
```

Comment:
It was about not having member/variables of type IDebugClient, but using DbgEng class that has properties that you need, creates client when it is needed, does conversion, etc... It is not that important.

---
In reply to: [70884085](https://github.com/southpolenator/WinDbgCs/pull/8#discussion_r70884085) [](ancestors = 70884085,69982378)